### PR TITLE
Arn 462 endpoint lates assessment rosh sections

### PR DESCRIPTION
**Pull request by adelaidaa**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/oasys/api/AssessmentSummaryDto.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import java.time.LocalDateTime
 data class AssessmentSummaryDto(
 
   @ApiModelProperty(value = "Assessment primary key (OASysSetPK)", example = "1234")
-  val assessmentId: Long? = null,
+  val assessmentId: Long,
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/oasys/api/AssessmentSummaryDto.kt:10]: oasyssetPk is a primary key so should ever be nullable

File: src/main/kotlin/uk/gov/justice/digital/oasys/api/AssessmentSummaryDto.kt [hunk 2]
```diff
@@ -46,25 +46,25 @@ data class AssessmentSummaryDto(
 
   companion object {
 
-    fun Collection<Assessment?>?.toAssessmentsSummaryDto(): Collection<AssessmentSummaryDto> {
-      return this?.filterNotNull()?.map { it.toAssessmentSummaryDto() }?.toSet().orEmpty()
+    fun Collection<Assessment>.toAssessmentsSummaryDto(): Collection<AssessmentSummaryDto> {
+      return this.map { it.toAssessmentSummaryDto() }?.toSet()
     }
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/oasys/api/AssessmentSummaryDto.kt:51]: remove the nullability as if we don't find any assessments we should be returning empty collection

File: src/main/kotlin/uk/gov/justice/digital/oasys/api/AssessmentSummaryDto.kt [hunk 3]
```diff
@@ -46,25 +46,25 @@ data class AssessmentSummaryDto(
...
 
     fun Assessment.toAssessmentSummaryDto(): AssessmentSummaryDto {
-      val assessmentVersion = this?.assessmentVersion
+      val assessmentVersion = this.assessmentVersion
       return AssessmentSummaryDto(
-        this?.oasysSetPk,
+        this.oasysSetPk,
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/oasys/api/AssessmentSummaryDto.kt:56]: oasyssetPk is a primary key so should ever be nullable

File: src/main/kotlin/uk/gov/justice/digital/oasys/jpa/entities/Assessment.kt [hunk 4]
```diff
@@ -18,7 +18,7 @@ data class Assessment(
 
   @Id
   @Column(name = "OASYS_SET_PK")
-  val oasysSetPk: Long? = null,
+  val oasysSetPk: Long,
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/oasys/jpa/entities/Assessment.kt:21]: oasyssetPk is a primary key so should ever be nullable

File: src/main/kotlin/uk/gov/justice/digital/oasys/jpa/repositories/AssessmentRepository.kt [hunk 5]
```diff
@@ -23,10 +23,11 @@ class AssessmentRepository constructor(entityManager: EntityManager) {
     filterAssessmentType: String? = null,
     filterVoided: Boolean? = null,
     filterAssessmentStatus: String? = null
-  ): Collection<Assessment>? {
+  ): Collection<Assessment> {
     val query = getAssessmentsQueryForOffender(offenderId)
     filterQuery(query, filterGroupStatus, filterAssessmentType, filterVoided, filterAssessmentStatus)
-    return query.fetch()
+    val assessments = query.fetch()
+    return assessments ?: emptySet()
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/oasys/jpa/repositories/AssessmentRepository.kt:30]: return empty list to stop propagating nullables

**Review by SWhitlam — APPROVED** [review]

**Pull request merged**